File: ccif/src/java/other/org/himalay/ccif/RatingRegion.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:33:05 EST 2016")

public  class RatingRegion extends BinStruct implements PublicBinMsg{




    // members variables
    // ratingRegionSetting
    public short ratingRegionSetting ;

        public RatingRegion () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize ratingRegionSetting
        
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read ratingRegionSetting
        {ratingRegionSetting=(short)(istream.readUnsignedByte()); retVal+=1;}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
    
                // write ratingRegionSetting
        ostream.writeByte(ratingRegionSetting); retVal +=1;
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("RatingRegion\n");
    dc.increaseIndent();
        int retVal= 0;
        // write ratingRegionSetting
        dc.indent();dc.getPs().println("ratingRegionSetting="+ratingRegionSetting+"(0x"+ Integer.toHexString(ratingRegionSetting) + ")" );
dc.decreaseIndent();
        return retVal;
    }


        // Getter for ratingRegionSetting
    //public short getRatingRegionSetting()
    //{
    //    return ratingRegionSetting ;
    //}

    
    // Setter for ratingRegionSetting
    //public void setRatingRegionSetting(short val)
    //{
    //    this.ratingRegionSetting= val;
    //}


    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/RatingRegion.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:17 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "RatingRegion.h"

#ifdef org_himalay_ccif_RatingRegion__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
RatingRegion::RatingRegion() {
    references= 0;


    // ratingRegionSetting
    ratingRegionSetting= (ui8)0;
}
// Destructor
RatingRegion::~RatingRegion() {
    // ratingRegionSetting

}

int RatingRegion::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // ratingRegionSetting
    {
        ratingRegionSetting=(istream.readUI8());
        retVal+=1;
    }
    return retVal;
}

int RatingRegion::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read ratingRegionSetting
    {
        ratingRegionSetting=(istream.readUI8());
        retVal+=1;
    }

    return retVal;
}

int RatingRegion::write(DataOutputStream& ostream) { // throws IOException


    int retVal= 0;
    // write ratingRegionSetting
    ostream.writeUI8(ratingRegionSetting);
    retVal +=1;

    return retVal;
}

int RatingRegion::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("RatingRegion\n");
    dc.increaseIndent();
    int retVal= 0;
    // write ratingRegionSetting
    dc.indent();
    dc.getPs().print("ratingRegionSetting=");
    dc.getPs().println((long)ratingRegionSetting);
    dc.decreaseIndent();
    return retVal;
}


// Getter for ratingRegionSetting
//ui8 RatingRegion::getRatingRegionSetting()
//{
//return this->ratingRegionSetting;
//}

// Setter for ratingRegionSetting
//void RatingRegion::setRatingRegionSetting(ui8 val)
//{
//this->ratingRegionSetting= val;
//}

int RatingRegion::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_RatingRegion__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::RatingRegion* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::RatingRegion* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code